File: src/main/java/bot/util/settings/SettingsInvoker.java
package bot.util.settings;

import bot.util.TextHandler;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

import java.util.LinkedHashMap;
import java.util.Map;

public class SettingsInvoker {
    private final TextHandler textHandler;
    private final Map<String, Setting> settingMap = new LinkedHashMap<>();

    public SettingsInvoker(TextHandler textHandler) {
        this.textHandler = textHandler;
    }

    public void add(String key, Setting setting) {
        settingMap.put(key, setting);
    }

    public void execute(String key, Update update) throws TelegramApiException {
        Setting setting = settingMap.get(key);
        if (setting != null) setting.execute(update);
        else {
            textHandler.text(key);
        }

    }
}

File: src/main/java/bot/util/settings/SettingsInvoker.kt
package bot.util.settings

import bot.util.TextHandler
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiException

class SettingsInvoker(private val textHandler: TextHandler) {

    private val settingMap: MutableMap<String, Setting> = LinkedHashMap()

    fun add(key: String, setting: Setting) {
        settingMap[key] = setting
    }

    fun execute(key: String, update: Update?) {
        val setting = settingMap[key]
        if (setting != null) setting.execute(update) else {
            textHandler.text(key, update)
        }
    }
}